dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/HttpHeadersInitializer.java

```diff
@@ -0,0 +1,263 @@
+    private static final String IMAGE = "image";
+    private static final String ACCEPT = "Accept";
+    private static final String CONTENT_DISPOSITION = "Content-Disposition";
+    private static final String CONTENT_DISPOSITION_FORMAT = "%s;filename=\"%s\"";
+    private static final String CACHE_CONTROL = "Cache-Control";
+
+    private int bufferSize = 1000000;
+
+    private static final long DEFAULT_EXPIRE_TIME = 60L * 60L * 1000L;
+
+    //no-cache so request is always performed for logging
+    private static final String CACHE_CONTROL_SETTING = "private,no-cache";
+
+    private BufferedInputStream inputStream;
+    private HttpServletRequest request;
+    private HttpServletResponse response;
+    private String contentType;
+    private String disposition;
+    private long lastModified;
+    private long length;
+    private String fileName;
+    private String checksum;
+
+    public HttpHeadersInitializer(final InputStream inputStream) {
+        //Convert to BufferedInputStream so we can re-read the stream
+        this.inputStream = new BufferedInputStream(inputStream);
+    }
+
+
+    public static HttpHeadersInitializer fromInputStream(InputStream inputStream) {
+        return new HttpHeadersInitializer(inputStream);
+    }
+
+    public HttpHeadersInitializer with(HttpServletRequest httpRequest) {
+        request = httpRequest;
+        return this;
+    }
+
+    public HttpHeadersInitializer with(HttpServletResponse httpResponse) {
+        response = httpResponse;
+        return this;
+    }
+
+    public HttpHeadersInitializer withLength(long length) {
+        this.length = length;
+        return this;
+    }
+
+    public HttpHeadersInitializer withFileName(String fileName) {
+        this.fileName = fileName;
+        return this;
+    }
+
+    public HttpHeadersInitializer withChecksum(String checksum) {
+        this.checksum = checksum;
+        return this;
+    }
+
+    public HttpHeadersInitializer withMimetype(String mimetype) {
+        this.contentType = mimetype;
+        return this;
+    }
+
+    public HttpHeadersInitializer withLastModified(long lastModified) {
+        this.lastModified = lastModified;
+        return this;
+    }
+
+    public HttpHeadersInitializer withBufferSize(int bufferSize) {
+        if (bufferSize > 0) {
+            this.bufferSize = bufferSize;
+        }
+        return this;
+    }
+    public HttpHeadersInitializer withDisposition(String contentDisposition) {
+        this.disposition = contentDisposition;
+        return this;
+    }
+
+    //TODO rename to initialiseHeaders
+    public HttpHeaders initialiseHeaders() throws IOException {
+
+        HttpHeaders httpHeaders = new HttpHeaders();
+        // Validate and process range -------------------------------------------------------------
+
+        log.debug("Content-Type : {}", contentType);
+        // Initialize response.
+        //TODO response.reset => Can be re-instated if we bump to 5.2.9
```

Comment:
@tdonohue-> (Heads up: it's been a while since we altered this and we seemed to have forgotten to take note of WHY exactly this was commented out)

One of the points mentioned in release tag [v5.2.9](https://github.com/spring-projects/spring-framework/releases/tag/v5.2.9.RELEASE) contains an [issue](https://github.com/spring-projects/spring-framework/issues/25501) related to the response.reset call, which seems to fit the description (And also jogged Raf's memory)

That being said: 
* The current spring versions seems 5.2.5 (which is released in [march](https://github.com/spring-projects/spring-framework/releases/tag/v5.2.5.RELEASE), around 8 months ago.)
* The last 5.2.x version is [5.2.11](https://github.com/spring-projects/spring-framework/releases/tag/v5.2.11.RELEASE), which is only a week old. 
* There are 2 courses that we can take imo
    * We upgrade it early to both get the advantage of all fixes that have been done up to now, as well as already getting a bunch of the upgrade done. (Upgrading later on might introduce MORE changes required). This DOES introduce the need for later upgrades as well if in the meantime, a big bug fix is released.
    * We wait until there is a more convenient time for this upgrade (less features/fixes in the making for example), and we upgrade then. That way, we only do one BIG upgrade at that time, instead of multiple smaller ones

Reply:
@jonas-atmire : I'm not sure I understand entirely...or at least I have a few more questions:

In terms of this `response.reset` call....what does this add to the code?  Is the code less stable in some way until we upgrade to v5.2.9 (or above)?  Put another way, what is the risk of *delaying* the upgrade until a later date.  Is the risk minimal, or do you two foresee issues here if we do not upgrade?

By default here, I'd rather **not** complicate this ticket and hold it back for an upgrade to Spring v.5.2.9.  But, it sounds like we *might* need another ticket here around upgrading to Spring v5.2.9 or above....and I'm not yet sure how to prioritize this, as I don't quite understand the risk of not upgrading immediately.

Reply:
@tdonohue , sorry for the confusion. I seemed to have linked to an incorrect issue previously.
This code does the following: 
* [Clears any data that exists in the buffer as well as the status code, headers.](https://docs.oracle.com/javaee/7/api/javax/servlet/ServletResponse.html#reset--)

I have reconvened with @Raf-atmire, who did the actual implementation of this upgrade.
During this implementation, this line of the code broke the feature when it was being triggered, so it was put in a comment temporarily.
(We do not actually see any risks with leaving this line out for the time being, so we can reduce the complication of this ticket, and holding it back because of a spring upgrade)

I have now re-enabled this line locally, and went through a quick test of downloading a content file created through the script processing, and this does NOT break the code anymore.
As such, I have re-enabled this line again.
If, during your testing, you see any issue with this come up, please let me know so we can indeed check if this is required for the time being or not.

Reply:
@jonas-atmire : Thanks for the more thorough testing/explanation.  I'm OK with leaving this commented out for now... if you and @Raf-atmire feel that the *ideal behavior* is that this `response.reset()` should be enabled, then I'd ask that we please create an issue ticket about re-enabling it once we upgrade to Spring v5.2.9 or above.  That way you can link/reference the ticket in your comment and say something like `//TODO response.reset => Can be re-instated once we upgrade to Spring 5.2.9, see issue #__`

That seems good enough to me. I'd just ask that we create a ticket as a reminder, as the TODO may end up forgotten otherwise.   I'll likely schedule the Spring upgrade for 7.1 or 7.2 since it seems low priority at this time.  Thanks!

Reply:
@tdonohue Sorry again for the back and forth.
I've subsequently commented out that part of the code again, and created a new issue referencing the piece of code.
https://github.com/DSpace/DSpace/issues/3056